src/raftstore/store/debug.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{error, result};
+use kvproto::debugpb::*;
+use rocksdb::DB as RocksDb;
+
+use raftstore::store::Engines;
+use raftstore::store::engine::Peekable;
+use storage::{CF_DEFAULT, CF_LOCK, CF_RAFT, CF_WRITE};
+
+pub type Result<T> = result::Result<T, Error>;
+
+quick_error!{
+    #[derive(Debug)]
+    pub enum Error {
+        InvalidArgument(msg: String) {
+            description(msg)
+            display("Invalid Argument {:?}", msg)
+        }
+        NotFound(msg: String) {
+            description(msg)
+            display("Not Found {:?}", msg)
+        }
+        Other(err: Box<error::Error + Sync + Send>) {
+            from()
+            cause(err.as_ref())
+            description(err.description())
+            display("{:?}", err)
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct Debugger {
+    engines: Engines,
+}
+
+impl Debugger {
+    pub fn new(engines: Engines) -> Debugger {
+        Debugger { engines }
+    }
+
+    pub fn get(&self, db: DB, cf: &str, key: &[u8]) -> Result<Vec<u8>> {
+        let db = try!(self.get_db(db));
+        let cf = try!(validate_cf(cf));
```

Comment:
Different db has different cfs, it may be simpler to let db itself to check if a cf is missing.